### PR TITLE
🛡️ Sentinel: [HIGH] Fix shell arithmetic injection

### PR DESCRIPTION
**Pull request by d-o-hub**
🛡️ Sentinel: [HIGH] Fix shell arithmetic injection

🚨 Severity: HIGH
💡 Vulnerability: Shell arithmetic contexts (like `$(( ))`, `(( ))`, and `[[ $A -lt $B ]]`) evaluate operands as expressions. If these operands contain variables that can be controlled by a user (e.g., via environment variables), they can be exploited to execute arbitrary commands using syntax like `a[$(id)0]`.
🎯 Impact: Arbitrary command execution through numeric configuration variables such as `SELF_FIX_LOOP_TIMEOUT`, `MAX_WASM_SIZE_BYTES`, etc.
🔧 Fix: Implemented strict numeric validation using `[[ "$var" =~ ^[0-9]+$ ]]` for all overridable numeric variables in the following scripts:
- `scripts/self-fix-loop.sh`
- `scripts/wasm_size_gate.sh`
- `scripts/lib/skill-validation.sh`
- `scripts/swarm-worktree-web-research.sh`
✅ Verification: Created reproduction scripts that confirmed the vulnerability was exploitable before the patch and successfully blocked after the patch. Ran the full `./scripts/quality_gate.sh` and it passed.

---
*PR created automatically by Jules for task [6876704381085425842](https://jules.google.com/task/6876704381085425842) started by @d-o-hub*

### CODE REVIEW

File: scripts/self-fix-loop.sh [hunk 1]
```diff
@@ -13,6 +13,15 @@ TIMEOUT="${SELF_FIX_LOOP_TIMEOUT:-1800}"
...
 AUTO_RESEARCH="${SELF_FIX_LOOP_AUTO_RESEARCH:-1}"
 STRICT_VALIDATION="${SELF_FIX_LOOP_STRICT_VALIDATION:-1}"
+
+# Security: Validate numeric configuration to prevent shell arithmetic injection
+for var in MAX_RETRIES TIMEOUT POLL_INTERVAL AUTO_RESEARCH STRICT_VALIDATION; do
+    if [[ ! "${!var}" =~ ^[0-9]+$ ]]; then
+        echo "Error: $var must be numeric" >&2
+        exit 1
+    fi
+done
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply scripts/self-fix-loop.sh]: @jules Address all PR review feedback completely and systematically.

Requirements:

* Analyze every review comment, requested change, CI signal, and inline discussion on the PR.
* Implement fixes with production-quality changes only — no temporary workarounds, commented-out code, or partial resolutions.
* Preserve existing behavior unless explicitly required by the feedback.
* Verify there are no regressions, edge-case failures, or broken integrations.

Validation workflow:

1. Re-run the full validation pipeline locally:

   * tests
   * lint
   * typecheck
   * build
   * benchmarks (if applicable)
2. Use real usage / dogfooding flows where possible instead of only unit tests.
3. Verify affected functionality through the actual CLI and end-to-end execution paths.
4. Confirm CI parity with local execution.
5. Inspect git diff carefully for accidental unrelated changes.

Git workflow:

* Commit all changes with clear conventional commit messages.
* Push all commits to the active PR branch.

PR verification (mandatory):

* Use GitHub CLI (`gh`) to inspect the PR after pushing.
* Verify:

  * all required checks are green or running successfully
  * no merge conflicts exist
  * PR state is `mergeable`
  * no unresolved review threads remain
  * branch is up to date with target branch if required

Use commands such as:

* `gh pr view --comments --json reviews,reviewThreads,mergeable,headRefName,statusCheckRollup`
* `gh pr checks`
* `gh pr status`

If the PR is not mergeable:

* identify the blocking issue
* resolve it completely
* re-verify until the PR is mergeable

Finally provide:

* concise summary of implemented fixes
* validation results
* CI/check status
* final mergeability status
* remaining risks or follow-ups (if any)
